terraform/a2ha-terraform/modules/keystoreopensearch/templates/opensearch-keystore.sh.tpl

```diff
@@ -27,31 +28,63 @@ SERVICE_UP_TIME=$(hab svc status  | awk  '{print $5}' | tail -1)
 
 # If you get following error, ===> `[: : integer expression expected`
 # try `until [[ "$SERVICE_UP_TIME" -gt 30 ]]` instead of `until [ "$SERVICE_UP_TIME" -gt 30 ]`
-if [ "${backup_config_s3}" == "true" ]; then
-
-export OPENSEARCH_PATH_CONF="/hab/svc/automate-ha-opensearch/config"
-if [ ! -f ${tmp_path}/$OS_SETUP_FILE ]; then
-max=15
-n=0
-until [ "$SERVICE_UP_TIME" -gt 30 -a $n -lt $max ]
-do
-  sleep 5
-  SERVICE_UP_TIME=$(hab svc status  | awk  '{print $5}' | tail -1)
-  echo "No services are loaded"
-  echo "Sleeping for 5 seconds"
-  n=$((n+1))
-done
-
-# Adding aws access and secret keys once all services are up
-
-  echo "Setting up keystore"
-  echo $OPENSEARCH_PATH_CONF
-  echo "${access_key}" | hab pkg exec "$OS_ORIGIN_NAME/$OS_PKG_NAME" opensearch-keystore add --stdin --force s3.client.default.access_key
-  echo "${secret_key}" | hab pkg exec "$OS_ORIGIN_NAME/$OS_PKG_NAME" opensearch-keystore add --stdin --force s3.client.default.secret_key
-  hab pkg exec "$OS_ORIGIN_NAME/$OS_PKG_NAME" opensearch-keystore list
-  sudo chown -RL hab:hab /hab/svc/automate-ha-opensearch/config/opensearch.keystore
-  hab pkg exec "$OS_ORIGIN_NAME/$OS_PKG_NAME" opensearch-keystore list
-  curl -k -X POST --cacert /hab/svc/automate-ha-opensearch/config/certificates/root-ca.pem --key /hab/svc/automate-ha-opensearch/config/certificates/admin-key.pem --cert /hab/svc/automate-ha-opensearch/config/certificates/admin.pem "https://127.0.0.1:${listen_port}/_nodes/reload_secure_settings?pretty"
-  sudo touch ${tmp_path}/$OS_SETUP_FILE
+if [ "${location}" == "s3" ]; then
```

Comment:
For old bundles, location value is not present. Should we consider the default value of the location as "s3" if the location is not provided? (Also this might make sure upgrade scenario works fine)

Reply:
handled in this PR: https://github.com/chef/automate/pull/8164/files
So resolving this comment